### ISSUE TITLE
Backport irrlicht_changes from Minetest

### ISSUE THREAD

**Issue by numberZero** [minetest/irrlicht#159]
`irrlicht_changes` were added before forking Irrlicht. Currently there is no point keeping them separate.

**Comment by sfan5** [comment]
This should be all that's left:
* https://github.com/minetest/minetest/tree/master/src/irrlicht_changes
* https://github.com/minetest/minetest/tree/master/src/gui

Not sure if it makes sense to move the `src/gui/` stuff to Irrlicht either since it might be too integrated to MT.
In fact it might not make sense to move _anything_ back since Irrlicht is supposed to merged into MT not the other way around.

**Comment by numberZero** [comment]
If the GUI stuff is all that is there, it may indeed make more sense to to move in the opposite direction: move *all* GUI stuff that MT still uses into MT, and drop GUI part here altogether.

**Comment by numberZero** [comment]
There is also [src/irrlichttypes.h](https://github.com/minetest/minetest/blob/master/src/irrlichttypes.h):
 * irr::core::roundingError<s16> ()
 * *_MIN, *_MAX

[src/irrlicht_changes](https://github.com/minetest/minetest/tree/master/src/irrlicht_changes) is GUI stuff only so it should likely merged into [src/gui](https://github.com/minetest/minetest/tree/master/src/gui).

> all GUI stuff that MT still uses

Looking at [src/gui](https://github.com/minetest/minetest/tree/master/src/gui), I suspect there is very little.

**Comment by Desour** [comment]
[irr_ptr](https://github.com/minetest/minetest/blob/master/src/irr_ptr.h) would be good to have in irrlicht.

**Comment by numberZero** [comment]
> [irr_ptr](https://github.com/minetest/minetest/blob/master/src/irr_ptr.h) would be good to have in irrlicht.

Makes sense. For the API I’d wait for The Merge but for internal use, it’s OK to start using it now IMO. My grep suggests there are 287 drop calls and 89 grab calls awaiting replacement.
